src/migration/MigrationExecutor.ts

```diff
@@ -232,28 +261,99 @@ export class MigrationExecutor {
                         {
-                            name: "id",
+                            name: COLUMN_ID,
                             type: this.connection.driver.normalizeType({type: this.connection.driver.mappedDataTypes.migrationId}),
                             isGenerated: true,
                             generationStrategy: "increment",
                             isPrimary: true,
                             isNullable: false
                         },
                         {
-                            name: "timestamp",
+                            name: COLUMN_TIMESTAMP,
                             type: this.connection.driver.normalizeType({type: this.connection.driver.mappedDataTypes.migrationTimestamp}),
                             isPrimary: false,
                             isNullable: false
                         },
                         {
-                            name: "name",
+                            name: COLUMN_NAME,
                             type: this.connection.driver.normalizeType({type: this.connection.driver.mappedDataTypes.migrationName}),
                             isNullable: false
                         },
+                        {
+                            name: COLUMN_HASH,
+                            type: this.connection.driver.normalizeType({type: this.connection.driver.mappedDataTypes.migrationHash}),
+                            isNullable: false,
+                            length: "40"
+                        }
                     ]
                 },
             ));
         }
+        const table = await queryRunner.getTable(this.migrationsTable);
+        const hashColumnExist = table!.columns.some(c => c.name.toLowerCase() === COLUMN_HASH);
+        if (!hashColumnExist) {
+            await queryRunner.addColumn(this.migrationsTable, new TableColumn({
+                name: COLUMN_HASH,
+                type: this.connection.driver.normalizeType({type: this.connection.driver.mappedDataTypes.migrationHash}),
+                isNullable: true,
+                length: "40",
+            }));
+
+            await this.updateHashes(queryRunner);
+
+            const {count} = await this.connection.manager.createQueryBuilder(queryRunner)
+            .select("count(*)", "count")
+            .from(this.migrationsTable, this.migrationsTableName)
+            .where("hash is null")
+            .getRawOne();
+
+            if (count === 0) {
+                await queryRunner.changeColumn(this.migrationsTable, COLUMN_HASH, new TableColumn({
+                    name: COLUMN_HASH,
+                    type: this.connection.driver.normalizeType({type: this.connection.driver.mappedDataTypes.migrationHash}),
+                    isNullable: false,
+                    length: "40",
+                }));
+            }
+        }
+
+        const createDateColumnExist = table!.columns.some(c => c.name.toLowerCase() === COLUMN_CREATE_DATE);
```

Comment:
Is `toLowerCase()` necesary here? It isn't obvious that COLUMN_CREATE_DATE have to be lowercased(and it's extracted to const so it would be easier to change in the future).

Reply:
Looks like my local tests will pass if I remove this check. I did this because Oracle keeps all the column names in uppercase, but I have no way to test that locally.